### PR TITLE
cpu, sam0 common: adapt periph-uart to use read-only bit-fields

### PR DESCRIPTION
**Pull request by smlng**
factored out of #7638, see discussion there.
~~Note: based on and includes #7638.~~

### CODE REVIEW

**Comment by smlng** [comment]
see also #7644

File: cpu/sam0_common/periph/uart.c [hunk 1]
```diff
@@ -119,10 +119,10 @@ int uart_init(uart_t uart, uint32_t baudrate, uart_rx_cb_t rx_cb, void *arg)
 void uart_write(uart_t uart, const uint8_t *data, size_t len)
 {
     for (size_t i = 0; i < len; i++) {
-        while (!(dev(uart)->INTFLAG.reg & SERCOM_USART_INTFLAG_DRE)) {}
+        while (!dev(uart)->INTFLAG.bit.DRE) {}
         dev(uart)->DATA.reg = data[i];
-        while (dev(uart)->INTFLAG.reg & SERCOM_USART_INTFLAG_TXC) {}
     }
+    while (!dev(uart)->INTFLAG.bit.TXC) {}
```
Comment (kaspar030) [inline cpu/sam0_common/periph/uart.c]: nitpick: this now contains #7638...

Comment (smlng) [reply cpu/sam0_common/periph/uart.c]: which is why I said above:

> Note: based on and includes #7638.

Comment (kaspar030) [reply cpu/sam0_common/periph/uart.c]: > based on and includes #7638.

No it doesn't.

#7638 is broken and this PR is just re-doing the same change. If we'd merge #7638, you'd have to fix the one line it changes when rebasing this PR.

Please @smlng, there is a reason why we keep PR's small and don't piggyback style changes.

Comment (smlng) [reply cpu/sam0_common/periph/uart.c]: okay factored fix out, now only replacing bit-fields here

**Comment by smlng** [comment]
rebased

**Comment by smlng** [comment]
@kaspar030, all good now?

**Comment by smlng** [comment]
@kaspar030 comments address (long time ago), care to ACK?

**Comment by smlng** [comment]
ping @kaspar030

**Comment by smlng** [comment]
@kaspar030 ACK or dismiss review - easy to have for todays HnA

**Comment by miri64** [comment]
@smlng @kaspar030's not at Hack'n'ACK today

**Comment by smlng** [comment]
and another month gone by ...

**Comment by miri64** [comment]
Better write him directly ;-). @kaspar030 told me a few weeks back that GitHub isn't showing him any notifications anymore.

**Comment by smlng** [comment]
knock ... knock @kaspar030 😉

**Comment by kaspar030** [comment]
I fixed my github notification problem by changing the email address. ;)

**Pull request merged**